### PR TITLE
Authentication: Addressing Debt

### PR DESCRIPTION
**Pull request by jleandroperez**
### Fix
In this PR we're addressing pending tech debt, in the AuthenticationViewController class.

### Test
Please smoke test Auth with Password + Code, and verify everything looks great!!

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Simplenote/AuthViewController.h [hunk 1]
```diff
@@ -52,6 +52,5 @@
 - (void)presentPasswordResetAlert;
 - (void)presentPasswordCompromisedAlert;
 - (void)presentUnverifiedEmailAlert;
-- (void)showAuthenticationErrorForCode:(NSInteger)responseCode responseString:(NSString *)responseString;
```
Comment (jleandroperez) [inline Simplenote/AuthViewController.h:55]: This was triggering a warning. I've missed dropping the forward declaration, when ported to Swift

File: Simplenote/AuthViewController+Swift.swift [hunk 2]
```diff
@@ -405,17 +397,9 @@ extension AuthViewController {
...
     }
-    
-    //TODO: Drop this method?
-    func presentMagicLinkRequestedView(email: String) {
-        guard let authWindowController else {
-            return
-        }
-        
-        authWindowController.switchToMagicLinkRequestedUI(email: email)
-    }
```
Comment (jleandroperez) [inline Simplenote/AuthViewController+Swift.swift:416]: Not in use any longer!

File: Simplenote/AuthViewController+Swift.swift [hunk 3]
```diff
@@ -367,9 +359,9 @@ extension AuthViewController {
 
         switch superView {
         case usernameField:
-            state.username = usernameField.stringValue
+            state.username = usernameField.stringValue.trimmingCharacters(in: .whitespacesAndNewlines)
         case passwordField:
-            state.password = passwordField.stringValue
+            state.password = passwordField.stringValue.trimmingCharacters(in: .whitespacesAndNewlines)
```
Comment (jleandroperez) [inline Simplenote/AuthViewController+Swift.swift:364]: Legacy `usernameText` and `passwordText` had this trimming (L97 + L102 in this PR)

File: Simplenote/AuthViewController+Swift.swift [hunk 4]
```diff
@@ -234,11 +231,6 @@ extension AuthViewController {
 // MARK: - Handlers
 //
 extension AuthViewController {
-
-    @IBAction
-    func switchToPasswordAuth(_ sender: Any) {
-        mode = AuthenticationMode.loginWithPassword
-    }
```
Comment (jleandroperez) [inline Simplenote/AuthViewController+Swift.swift:241]: This one wasn't in use

File: Simplenote/AuthViewController+Swift.swift [hunk 5]
```diff
@@ -93,18 +94,14 @@ extension AuthViewController {
...
 
     @objc
     var passwordText: String {
-        passwordField.stringValue?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
+        state.password
     }
     
-    var authWindowController: AuthWindowController? {
-        view.window?.windowController as? AuthWindowController
-    }
```
Comment (jleandroperez) [inline Simplenote/AuthViewController+Swift.swift:106]: 👋 too!!

**Review by charliescheer — APPROVED** [review]

**Comment by jleandroperez** [comment]
TY!! 🔥

**Pull request merged**